### PR TITLE
Fixes for Windows compatibility (backpatch to 3.7.x)

### PR DESCRIPTION
**Pull request by evanweible-wf**
There are several paths that we're using in our executable and in the run script that we generate and we need to construct them with the appropriate APIs from `package:path` to ensure they work when running on Windows. This PR does that.

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: test/utils/dart_dev_paths_test.dart [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        expect(paths.cache('sub/path'), r'.dart_tool\dart_dev\sub\path');
+      });
+
+      test('config', () {
+        expect(paths.config, r'tool\dart_dev\config.dart');
+      });
+
+      test('configFromRunScriptForDart', () {
+        expect(paths.configFromRunScriptForDart,
+            r'../../tool/dart_dev/config.dart');
```
Comment (todbachman-wf) [inline test/utils/dart_dev_paths_test.dart:61]: Are the path separators for a relative path really not backslashes in windows? If so, how inconsistent.

Comment (todbachman-wf) [reply test/utils/dart_dev_paths_test.dart:61]: Turns out this is a path that will be used as an import in a .dart file. Dart always uses forward slashes regardless of the platform. So, nothing to do here.

**Review by todbachman-wf — APPROVED** [review]

**Comment by evanweible-wf** [comment]
QA +1
- [x] Includes cherry-picked commits from https://github.com/Workiva/dart_dev/pull/386#issuecomment-1228948110 which also includes running ddev tests against Windows.
- [x] Had a Windows user manually verify that this fixes the issue with the generated run script.

@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**